packages/grid/_modules_/grid/components/GridCell.tsx

```diff
@@ -102,6 +101,21 @@ export const GridCell: React.FC<GridCellProps> = React.memo((props) => {
     [publish, publishClick],
   );
 
+  const onDragEnter = React.useCallback((event) => apiRef!.current.onColItemDragEnter(event), [
```

Comment:
this onColItem... should be refactored and will be removed
#1201

Reply:
I think we still need to keep the APIs because they are used in the header cells but we will need to add handlers on the grid component itself that call those APIs to satisfy the requirement from https://github.com/mui-org/material-ui-x/issues/1201

Reply:
apiRef should not be trigger for events.

There should not be any method like `handleDragEnd ` on it. You can subscribe to an handler using `subscribeEvent` but not just put them there. That's not its purpose.

Only api methods that can be used to retrieve data, or manipulate the grid. 
Something like `setColumnIndex(field, index)` is appropriate in this case.
So users can reset the index outside the drag action.